### PR TITLE
Feature/damage repair system

### PR DESCRIPTION
**Pull request by lucasvdiepen**

### CODE REVIEW

File: Assets/Scripts/Framework/DamageRepairManagement/Damageable.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        /// <summary>
+        /// Invoked when the health is changed.
+        /// </summary>
+        public Action<int> OnHealthChanged;
+
+        [SerializeField]
+        private int _maxHealth;
+
+        [SerializeField]
+        private bool _canBeBelowZero;
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/DamageRepairManagement/Damageable.cs]:
```suggestion
        private bool _canHaveNegativeHealth;
```

File: Assets/Scripts/Framework/Gameplay/Interactions/Interactables/ShelfInteractable.cs [hunk 2]
```diff
@@ -80,5 +100,11 @@ private bool GrabFromShelf(Interactor interactor)
             interactor.UpdateInteractingTarget(interactor.Owner, targetInteractable, true);
             return targetInteractable.InteractAsServer(interactor);
         }
+
+        [Server]
+        private void OnHealthChanged(int health)
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/Gameplay/Interactions/Interactables/ShelfInteractable.cs:106]:
```suggestion
        private void OnHealthChanged(int health)
               => _heldGrabbableDamageable.CanTakeDamage = health <= 0;
```

**Review by PiterGroot — CHANGES_REQUESTED** [review]
Looks very nice.

**Review by Joost-Robat — COMMENTED** [review]
Looks good.

**Review by PiterGroot — APPROVED** [review]

**Review by Joost-Robat — APPROVED** [review]
Looks good, ready for merge IMO.

**Review by Joost-Robat — APPROVED** [review]
Looks good, ready for merge.

**Pull request merged**